File: backend/app/schemas/statistics.py
from pydantic import BaseModel


class StatisticsRequest(BaseModel):
    file_ids: list[int]


class DigitStatistics(BaseModel):
    counts: dict[str, int]


class FileStatistics(BaseModel):
    id: int

    name: str

    statistics: DigitStatistics


class StatisticsResponse(BaseModel):
    total: DigitStatistics

    files: list[FileStatistics]

File: frontend/src/types/statistics.ts
export interface DigitStatistics {

    counts: Record<string, number>;

}



export interface FileStatistics {

    id: number;

    name: string;

    statistics: DigitStatistics;

}



export interface StatisticsResponse {

    total: DigitStatistics;

    files: FileStatistics[];

}